### PR TITLE
Update hover color for lounge access icon

### PR DESCRIPTION
**Pull request by hungvu193**
### Details

Update hover color for lounge access icon
### Fixed Issues

$ https://github.com/Expensify/App/issues/22766
PROPOSAL: https://github.com/Expensify/App/issues/22766#issuecomment-1633483867

### Tests

1. Click on the settings (profile icon)
2. Hover over various buttons, observe color change of the icon.
3. Click Profile.
4. Hover over personal details, observe color change of the icon.
5. Hover over or long pressing Lounge Access.
6. Verify that LoungeAccess icon turn into white when hovering.
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests.
### QA Steps

1. Click on the settings (profile icon)
2. Hover over various buttons, observe color change of the icon.
3. Click Profile.
4. Hover over personal details, observe color change of the icon.
5. Hover over or long pressing Lounge Access.
6. Verify that LoungeAccess icon turn into white when hovering.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/16502320/0f01673c-8f3a-42a7-a4eb-f37798d3f3a4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/16502320/6e192a48-0536-413e-8914-65b5f69918e5

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/16502320/1e9d1610-0a04-41c1-8ece-9577930e0869

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/16502320/0f01673c-8f3a-42a7-a4eb-f37798d3f3a4

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/16502320/5ef77ac2-e949-422a-9d3b-b3061f70975c

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/16502320/777f9747-db3e-4915-a518-6db33afa8ff2

</details>

### CODE REVIEW

File: src/components/Icon/svgs/LoungeAccessIcon.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import PropTypes from 'prop-types';
+import themeColors from '../../../styles/themes/default';
+import variables from '../../../styles/variables';
+
+const propTypes = {
+    /** The width of the icon. */
+    width: PropTypes.number,
+
+    /** The height of the icon. */
+    height: PropTypes.number,
```
Comment (situchan) [inline src/components/Icon/svgs/LoungeAccessIcon.js:12]: These props are unused are unused and can be removed

File: src/components/Icon/index.js [hunk 2]
```diff
@@ -58,6 +66,8 @@ class Icon extends PureComponent {
                             width={width}
                             height={height}
                             fill={this.props.fill}
+                            isHovered={this.props.isHovered}
+                            isPressed={this.props.isPressed}
```
Comment (situchan) [inline src/components/Icon/index.js:70]:
```suggestion
                            hovered={this.props.isHovered.toString()}
                            pressed={this.props.isPressed.toString()}
```

File: src/components/Icon/index.js [hunk 3]
```diff
@@ -73,6 +83,8 @@ class Icon extends PureComponent {
                     width={width}
                     height={height}
                     fill={this.props.fill}
+                    isHovered={this.props.isHovered}
+                    isPressed={this.props.isPressed}
```
Comment (situchan) [inline src/components/Icon/index.js:87]:
```suggestion
                    hovered={this.props.isHovered.toString()}
                    pressed={this.props.isPressed.toString()}
```

File: src/components/Icon/svgs/LoungeAccessIcon.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    /** The fill color for the icon. Can be hex, rgb, rgba, or valid react-native named color such as 'red' or 'blue'. */
+    fill: PropTypes.string,
+
+    /** Is icon hovered */
+    isHovered: PropTypes.bool,
```
Comment (situchan) [inline src/components/Icon/svgs/LoungeAccessIcon.js:18]:
```suggestion
    hovered: PropTypes.string,
```

File: src/components/Icon/svgs/LoungeAccessIcon.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    /** Is icon pressed */
+    isPressed: PropTypes.bool,
```
Comment (situchan) [inline src/components/Icon/svgs/LoungeAccessIcon.js:21]:
```suggestion
    pressed: PropTypes.string,
```

File: src/components/Icon/svgs/LoungeAccessIcon.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+const defaultProps = {
+    width: variables.iconSizeNormal,
+    height: variables.iconSizeNormal,
+    fill: themeColors.icon,
+    isHovered: false,
+    isPressed: false,
```
Comment (situchan) [inline src/components/Icon/svgs/LoungeAccessIcon.js:29]:
```suggestion
    hovered: 'false',
    pressed: 'false',
```

File: src/components/Icon/svgs/LoungeAccessIcon.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        >
+            <G>
+                <Path
+                    className="st0"
+                    d="M11,24v-2c0-1.1,0.9-2,2-2s2,0.9,2,2c0,0.6,0.4,1,1,1h8c0.6,0,1-0.4,1-1c0-1.1,0.9-2,2-2s2,0.9,2,2v2 c0,0.4-0.1,0.8-0.3,1.1C28.1,26,27,26.9,27,28h-2v-2H15v2h-2c0-1.1-1.1-2-1.7-2.9C11.1,24.8,11,24.4,11,24z"
+                />
+            </G>
+            <G>
+                <Path
+                    fill={props.isHovered || props.isPressed ? props.fill : themeColors.starDefaultBG}
```
Comment (situchan) [inline src/components/Icon/svgs/LoungeAccessIcon.js:56]:
```suggestion
                    fill={props.hovered === 'true' || props.pressed === 'true' ? props.fill : themeColors.starDefaultBG}
```

**Review by situchan — CHANGES_REQUESTED** [review]
Requesting these changes because of these warnings:

Screenshot 2023-08-11 at 1 52 04 PM
Screenshot 2023-08-11 at 2 04 34 PM

**Comment by situchan** [comment]
@hungvu193 star not changing to white on hover. can you investigate?

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/108292595/2f21cde0-7525-4778-967a-cd8c13109ab4



</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://github.com/Expensify/App/assets/108292595/a524105c-64a5-4eed-bd7c-f971ba61cf14



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://github.com/Expensify/App/assets/108292595/6fcd4749-9434-4833-a0a8-dfb66b6e9532



</details>

<details>
<summary>Desktop</summary>


https://github.com/Expensify/App/assets/108292595/e0960836-9942-4efd-9ea5-91ff079d97d0



</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/108292595/b0cb03fd-e02e-4318-aa1d-710144a5ae49



</details>

<details>
<summary>Android</summary>


https://github.com/Expensify/App/assets/108292595/bf4e8756-a938-4fda-b513-641a968be930



</details>

**Review by situchan — APPROVED** [review]
LGTM :shipit:

**Review by aldo-expensify — APPROVED** [review]
Thanks for the ping!

It looks good to me

**Pull request merged**